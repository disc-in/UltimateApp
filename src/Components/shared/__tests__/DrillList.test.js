import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import { createDrill } from '../../../Fixtures/TestFixtures';
import store from '../../../Store/testStore';

import ConnectedDrillList, { DrillList } from '../DrillList';

describe('<DrillList />', () => {
  const drill = createDrill({ id: 1 });
  const customDrill = createDrill({ id: 2, custom: true });
  const drills = [drill, customDrill];

  it('renders correctly', async () => {
    const { toJSON } = await waitFor(() =>
      render(
        <Provider store={store}>
          <ConnectedDrillList drillsToDisplay={drills} />
        </Provider>,
      ),
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('can edit a custom drill', async () => {
    const navigate = jest.fn();
    const navigation = { navigate };

    const { getByTestId } = render(
      <Provider store={store}>
        <DrillList drillsToDisplay={drills} navigation={navigation} />
      </Provider>,
    );

    fireEvent.press(getByTestId('editButton'));
    await waitFor(() => expect(navigate).toBeCalledWith('DrillEditorPage', { currentDrill: customDrill }));
  });

  it('can delete a custom drill', async () => {
    const deleteDrill = jest.fn();

    const { getByTestId } = render(
      <Provider store={store}>
        <DrillList drillsToDisplay={drills} deleteDrill={deleteDrill} />
      </Provider>,
    );

    fireEvent.press(getByTestId('deleteButton'));
    await waitFor(() => expect(deleteDrill).toBeCalledWith(customDrill.id));
  });
});
