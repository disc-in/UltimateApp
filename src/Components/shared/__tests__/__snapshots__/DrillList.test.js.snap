// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrillList /> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 40,
    }
  }
  data={
    Array [
      Object {
        "author": "Author",
        "description": "Description of the drill",
        "durationInMinutes": 10,
        "equipment": "Equipment needs for the drill",
        "equipmentLabel": "None",
        "goals": Array [
          "Full body",
        ],
        "id": 1,
        "image": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
        "intensity": "Low",
        "level": "Beginner",
        "minimalPlayersNumber": 1,
        "seasonTiming": "Off-season",
        "steps": Array [
          Object {
            "subtitle": "This is a subtitle",
            "title": "Warmup",
            "vimeoId": "462695757",
          },
          Object {
            "subtitle": "This is a subtitle",
            "title": "Run !",
            "vimeoId": "462695757",
          },
        ],
        "title": "Fitness Drill Title",
        "type": "fitness",
        "visibleInList": true,
      },
      Object {
        "author": "Author",
        "custom": true,
        "description": "Description of the drill",
        "durationInMinutes": 10,
        "equipment": "Equipment needs for the drill",
        "equipmentLabel": "None",
        "goals": Array [
          "Full body",
        ],
        "id": 2,
        "image": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
        "intensity": "Low",
        "level": "Beginner",
        "minimalPlayersNumber": 1,
        "seasonTiming": "Off-season",
        "steps": Array [
          Object {
            "subtitle": "This is a subtitle",
            "title": "Warmup",
            "vimeoId": "462695757",
          },
          Object {
            "subtitle": "This is a subtitle",
            "title": "Run !",
            "vimeoId": "462695757",
          },
        ],
        "title": "Fitness Drill Title",
        "type": "fitness",
        "visibleInList": true,
      },
    ]
  }
  deleteDrill={[Function]}
  disableVirtualization={false}
  drillsToDisplay={
    Array [
      Object {
        "author": "Author",
        "description": "Description of the drill",
        "durationInMinutes": 10,
        "equipment": "Equipment needs for the drill",
        "equipmentLabel": "None",
        "goals": Array [
          "Full body",
        ],
        "id": 1,
        "image": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
        "intensity": "Low",
        "level": "Beginner",
        "minimalPlayersNumber": 1,
        "seasonTiming": "Off-season",
        "steps": Array [
          Object {
            "subtitle": "This is a subtitle",
            "title": "Warmup",
            "vimeoId": "462695757",
          },
          Object {
            "subtitle": "This is a subtitle",
            "title": "Run !",
            "vimeoId": "462695757",
          },
        ],
        "title": "Fitness Drill Title",
        "type": "fitness",
        "visibleInList": true,
      },
      Object {
        "author": "Author",
        "custom": true,
        "description": "Description of the drill",
        "durationInMinutes": 10,
        "equipment": "Equipment needs for the drill",
        "equipmentLabel": "None",
        "goals": Array [
          "Full body",
        ],
        "id": 2,
        "image": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
        "intensity": "Low",
        "level": "Beginner",
        "minimalPlayersNumber": 1,
        "seasonTiming": "Off-season",
        "steps": Array [
          Object {
            "subtitle": "This is a subtitle",
            "title": "Warmup",
            "vimeoId": "462695757",
          },
          Object {
            "subtitle": "This is a subtitle",
            "title": "Run !",
            "vimeoId": "462695757",
          },
        ],
        "title": "Fitness Drill Title",
        "type": "fitness",
        "visibleInList": true,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "height": 80,
            "marginBottom": 20,
            "opacity": 1,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(0,0,0)",
              "borderRadius": 5,
              "height": 80,
              "justifyContent": "center",
              "marginRight": 10,
              "padding": 10,
              "width": 80,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 80,
                  "width": 80,
                },
                Object {
                  "borderRadius": 5,
                  "opacity": 0.5,
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#e5e5e5",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            10
          </Text>
          <Text
            style={
              Object {
                "color": "#e5e5e5",
                "fontSize": 14,
                "textAlign": "center",
              }
            }
          >
            min
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "padding": 5,
              "paddingBottom": 10,
              "paddingRight": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#A6A6A6",
                "flex": 2,
                "fontSize": 14,
              }
            }
          >
            Author
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "flex": 3,
                "flexWrap": "wrap",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Fitness Drill Title
          </Text>
          <Text
            style={
              Object {
                "color": "#A6A6A6",
                "flex": 2,
                "fontSize": 14,
              }
            }
          >
            Full body
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2a9e9122",
            "borderRadius": 5,
            "flexDirection": "row",
            "height": 80,
            "marginBottom": 20,
            "opacity": 1,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(0,0,0)",
              "borderRadius": 5,
              "height": 80,
              "justifyContent": "center",
              "marginRight": 10,
              "padding": 10,
              "width": 80,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://www.dialysistech.org/wp-content/uploads/2019/06/fitness.jpg",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 80,
                  "width": 80,
                },
                Object {
                  "borderRadius": 5,
                  "opacity": 0.5,
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#e5e5e5",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            10
          </Text>
          <Text
            style={
              Object {
                "color": "#e5e5e5",
                "fontSize": 14,
                "textAlign": "center",
              }
            }
          >
            min
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "padding": 5,
              "paddingBottom": 10,
              "paddingRight": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#A6A6A6",
                "flex": 2,
                "fontSize": 14,
              }
            }
          >
            Author
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "flex": 3,
                "flexWrap": "wrap",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Fitness Drill Title
          </Text>
          <Text
            style={
              Object {
                "color": "#A6A6A6",
                "flex": 2,
                "fontSize": 14,
              }
            }
          >
            Full body
          </Text>
        </View>
        <View
          style={
            Object {
              "justifyContent": "space-evenly",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="editButton"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#2a9e91",
                    "fontSize": 22,
                  },
                  undefined,
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="deleteButton"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#2a9e91",
                    "fontSize": 22,
                  },
                  undefined,
                  Object {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
